src/App.js

```diff
@@ -0,0 +1,172 @@
+import BoardList from './components/BoardList';
+import CardList from './components/CardList';
+import NewCardForm from './components/NewCardForm';
+import SelectedBoard from './components/SelectedBoard';
+import NewBoardForm from './components/NewBoardForm'
+import { useEffect, useState } from 'react';
+import axios from 'axios';
+import './App.css';
+
+function App() {
+  const kBaseURLBoards = 'http://127.0.0.1:5000/boards'
+  const kBaseURLCards = 'http://127.0.0.1:5000/cards'
+
+  const [boardData, setBoardData] = useState([]);
+  const [cardData, setCardData] = useState([]);
+  const [selectedBoard, setSelectedBoard] = useState({});
+
+  const fetchBoards = () => {
+    axios
+    .get(kBaseURLBoards)
+    .then((res) => {
+      setBoardData(res.data)
+    })
+    .catch((err) => console.log(err))
+  };
+
+  useEffect(() => {
+    fetchBoards();
+  }, []);
+
+  const handleSubmit = (data) => {
+    axios
+    .post(kBaseURLCards, data)
+    .then((res) => {
+      console.log(res.data);
+      linkCardToBoard(res.data.id)
+    })
+    .catch((err) => console.log(err));
+  };
+
+  const linkCardToBoard = (data) => {
+    const newCardId = {id: data}
+
+    axios
+    .post(`${kBaseURLBoards}/${selectedBoard.id}/cards`, newCardId)
+    .then((res) => {
+      console.log(res.data);
+      onBoardSelect(selectedBoard.id);
+    })
+    .catch((err) => console.log(err));
+  };
+
+  const submitForm = (newBoard) => {
+    axios.post('http://127.0.0.1:5000/boards', newBoard)
+    .then((response) => {
+      setBoardData((prevBoardData) => 
+        [...prevBoardData, newBoard])
+      console.log(response);
+      fetchBoards();
```

Comment:
👀 Since you're re-retrieving the board list here, there's no need to set the board data here as well. If you _did_ want to just graft the new board data into the board list (without needing to re-fetch the whole list), be sure to use the result returned from the `post` call, as that board data would have the new board's id (which would not be present in `newBoard`, as that's only the data that came from the new board form).